gengokit/httptransport/httptransport.go

```diff
@@ -260,9 +260,19 @@ err = json.Unmarshal([]byte({{.LocalName}}Str), {{.LocalName}})`
 		// All repeated args of any type are represented as slices, and bare
 		// assignments to a slice accept a slice as the rvalue. As a result,
 		// LocalName will be declared as a slice, and json.Unmarshal handles
-		// everything else for us.
+		// everything else for us. Addititionally, if a type is a Base type and
+		// is repeated, we first attempt to unmarshal the string we're
+		// provided, and if that fails, we try to unmarshal the string
+		// surrounded by square brackets. If THAT fails, then the string does
+		// not represent a valid JSON string and an error is returned.
 		repeatedUnmarshalTmpl := `
 var {{.LocalName}} {{.GoType}}
+{{- if and (and .IsBaseType .Repeated) (not (Contains .GoType "[]byte"))}}
+err = json.Unmarshal([]byte({{.LocalName}}Str), &{{.LocalName}})
```

Comment:
This is probably still cheaper than reflecting twice.

As this happens per request it would be good to be efficient here, especially if someone is sending a lot of json requests without the `[]`'s.

Though switch a switch in your condition will add even more speed!

```
// Short circuit if v != null
if len(v) > 0 && v != null && !(v[0] == '[' || v[len(v)-1] == ']') {
```